### PR TITLE
Adding rule 4.3.3 related to null-checks

### PR DESCRIPTION
**Pull request by akuleshov7**
### What's done:
1) Adding rule that prohibits explicit null-checks (like a == null)
2) Autofix will be added in separate PR

### CODE REVIEW

**Review by kentr0w — COMMENTED** [review]
What about `equals`? Should this rule trigger on this case?

no, only `===`, `==`, `!=`, `!===`

**Review by kentr0w — APPROVED** [review]
lgtm

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    private var isFixMode: Boolean = false
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        if (node.elementType == CONDITION) {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt:30]: Shouldn't we also check number of conditional branches and handle only simple conditions?

**Review by petertrr — APPROVED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt (hunk 1 above)
Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt:30]: https://github.com/cqfn/diKTat/issues/529

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.psi.KtBinaryExpression
+
+/**
+ * This rule check and fixes explicit null checks (explicit comparison with `null`)
+ * There are several code-structures that can be used in Kotlin to avoid null-checks. For example: `?:`,  `.let {}`, `.also {}`, e.t.c
+ */
+class NullChecksRule(private val configRules: List<RulesConfig>) : Rule("null-checks") {
+    private var isFixMode: Boolean = false
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
```
[the remark this answers is not in the document]
Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullChecksRule.kt:22]: need to fix it everywhere in code finally

**Pull request merged**